react/components/Checkbox/index.js

```diff
@@ -79,7 +79,10 @@ class Checkbox extends PureComponent {
           </div>
           <input
             checked={checked}
-            ref={forwardedRef}
+            ref={elem => {
+              elem && (elem.indeterminate = !checked && partial)
```

Comment:
can we do this in `componentDidMount`/`componentDidUpdate`? feels a bit hacky to do this in the ref callback, which I don't think is supposed to mutate the DOM node

Reply:
there is no difference beyond the number of lines of code

Reply:
I agree with @lucasecdb ... I have never seen someone mutate a DOM node in the ref callback. It really looks like it's a job for the callbacks he mentioned.

Reply:
the problem is not the mutation, because regardless of where it is being done (inline or in its life cycle) it will be a mutation. According to gaearon the problem inline is apparently related to server rendering. So i will change to use DidMount and DidUpdate!